core/client/hdfs/src/main/java/alluxio/hadoop/LocalCacheFileSystem.java

```diff
@@ -112,10 +121,8 @@ public FSDataInputStream open(Path path, int bufferSize) throws IOException {
     if (mCacheManager == null) {
       return mExternalFileSystem.open(path, bufferSize);
     }
-    return new FSDataInputStream(new HdfsFileInputStream(
-        new LocalCacheFileInStream(new AlluxioURI(path.toString()),
-            OpenFilePOptions.getDefaultInstance(), mExternalFileSystemWrapper, mCacheManager),
-        statistics));
+    URIStatus status = HadoopUtils.toAlluxioUriStatus(mExternalFileSystem.getFileStatus(path));
+    return open(status, bufferSize);
```

Comment:
Do we only need 2 of these three methods? ie. the interface `open(Path, int)` and implementation `open(URIStatus, int)`? We should always use the hadoop file opener if cache is not enabled?

Also, buffer size is only respected if the cache is not enabled?

Reply:
> Do we only need 2 of these three methods? ie. the interface open(Path, int) and implementation open(URIStatus, int)? We should always use the hadoop file opener if cache is not enabled?

correct, i removed the 3rd `open` method with `FileOpener`. originally I thought I needed in Presto land, but it turns out not the case (see draft PR https://github.com/prestodb/presto/pull/15344/files)

> Also, buffer size is only respected if the cache is not enabled?

Correct